tests/subsys/tracing/src/main.c

```diff
@@ -0,0 +1,258 @@
+
+		k_sem_give(other_sem);
+	}
+}
+
+/* spawn a thread to generate more tracing data about thread switch. */
+void create_handle_thread(void)
+{
+	k_thread_create(&thread, thread_stack, STACK_SIZE,
+		thread_handle, NULL, &thread2_sem, &thread1_sem,
+		K_PRIO_PREEMPT(0), K_INHERIT_PERMS,
+		K_NO_WAIT);
+
+	thread_handle(NULL, &thread1_sem, &thread2_sem);
+}
+
+/**
+ * we need to use external tools to handle tracing data
+ * which produced by qemu/posix platform with different
+ * backends. and use popen api to execute the tools, so
+ * below test cases only avalible on posix arch.
+ */
+#ifdef CONFIG_ARCH_POSIX
+#include <unistd.h>
+#include <stdlib.h>
+#define MAX_BYTES 1024
+char result_buf[MAX_BYTES];
+
+/**
+ * @brief Test if tracing support thread, ISR, idle thread operation.
+ *
+ * @details
+ * - The consumer of the tracing data shall be able to identify and show
+ * the current phase of operation (thread, ISR, idle thread, etc.)
+ * based on the provided trace data.
+ *
+ * @ingroup subsys_tracing_tests
+ */
+void test_tracing_function_posix(void)
+{
+#if !defined(CONFIG_TRACING_BACKEND_POSIX) || defined(CONFIG_TRACING_CTF)
+	ztest_test_skip();
+#endif
+	FILE * fp;
+	bool thread_switched = false, isr_entered = false, idle_entered = false;
+
+	fp = fopen("channel0_0", "r");
+	if (fp == NULL) {
+		zassert_false(1, "fail to open tracing data file.");
+	}
+
+	while (fgets(result_buf, MAX_BYTES, fp) != NULL) {
+		/* remove space line */
+		if ('\n' == result_buf[strlen(result_buf)-1]) {
+			result_buf[strlen(result_buf)-1] = '\0';
+		}
+		if (strstr(result_buf, "sys_trace_thread_switched") != NULL) {
+			thread_switched = true;
+		}
+		if (strstr(result_buf, "isr_enter") != NULL) {
+			isr_entered = true;
+		}
+		if (strstr(result_buf, "idle") != NULL) {
+			idle_entered = true;
+		}
+	}
+
+	zassert_true(thread_switched, "doesn't find thread_switched.");
+	zassert_true(isr_entered, "doesn't find isr_entered.");
+	zassert_true(idle_entered, "doesn't find idle_entered.");
+
+	fclose(fp);
+}
+
+/**
+ * @brief Test if tracing support CTF.
+ *
+ * @details
+ * - The tracing system if support the common tracing format (CTF).
+ *
+ * @ingroup subsys_tracing_tests
+ */
+void test_tracing_function_posix_ctf(void)
+{
+#ifndef CONFIG_TRACING_CTF
+	ztest_test_skip();
+#endif
+	FILE * fp;
+
+	fp = popen("file channel0_0", "r");
+	if (fp == NULL) {
+		zassert_false(1, "fail to open tracing data file.");
+	}
+
+	/* CTF file type is data, not ASCII text
+	 * so we check if the file type is data.
+	 */
+	while (fgets(result_buf, MAX_BYTES, fp) != NULL) {
+		/* remove space line */
+		if ('\n' == result_buf[strlen(result_buf)-1]) {
+			result_buf[strlen(result_buf)-1] = '\0';
+		}
+		zassert_true(strcmp(result_buf, "channel0_0: data") == 0,
+				"CTF should be data format.");
+	}
+
+	pclose(fp);
+}
+
+/**
+ * @brief Test if tracing product redundant code when tracing disabled.
+ *
+ * @details
+ * - All tracing calls shall not be producing code when
```

Comment:
this sounds like a requirement, not a detail of what the test does.

Reply:
I will add detail infor,  thanks.